docs/concepts/explore-available-balancer-pools/reclamm-pool/reclamm-pool-math.md

```diff
@@ -0,0 +1,258 @@
+2. Isolate $V_b$ in the centeredness formula: $V_b = \frac{R_bV_acenteredness}{R_a}$ (if $\frac{R_a}{R_b} > \frac{V_a}{V_b}$, `centeredness` goes in the denominator)
+3. Replace $V_a$ formula in the $V_b$ formula, and the result will be:
+
+$V_b^2 (Q_{0_{new}} - 1) - V_b (R_b(1 + centeredness)) - R_b^2centeredness = 0$
+
+4. Resolve the formula above with Bhaskara to find the value of $V_b$, then replace the value of $V_b$ in the $V_a$ formula.
+
+## Calculation of the Virtual Balances
+
+### When initializing the pool
+
+During pool creation, pass minimum and maximum prices of A ($P_{a_{min}}$ and $P_{a_{max}}$), target price $P_{target}$. The contract also needs to assume an arbitrary $R_{a_{max}}'$, which we suggest should be `1000 * FixedPoint.ONE`.
+
+This allows the pool, during creation, to calculate the virtual balances as:
+
+$V_{a}' = \frac{R_{a_{max}}'}{Q_0 - 1}$
+
+$V_b' = P_{a_{min}}(R_{a_{max}}' + V_a')$
+
+Note, in the equation above, that $Q_0 = \sqrt\frac{P_{a_{max}}}{P_{a_{min}}}$. Also, note that $V_a'$ and $V_b'$ are scaled according to $R_{a_{max}}'$. During the pool initialization, these parameters will be scaled according to the real balances.
+
+Calculate the theoretical balances $R_a'$ and $R_b'$ for the default $R_{a_{max}}'$.
+
+$R_b' = \sqrt{P_{target}V_b'V_a'Q_0} - V_b'$
+
+$R_a' = \frac{R_b' + V_b' - V_a'P_{target}}{P_{target}}$
+
+Calculate the pool centeredness using the parameters above, and check if the pool centeredness is above the margin. If not, reverts.
+
+### Proportion Getter
+
+Between pool creation and initialization, the user must be able to calculate the correct token proportions needed to ensure the target price is respected. This proportion is $\frac{R_b'}{R_a'}$.
+
+### Pool initialization
+
+To initialize the pool, we receive the real balances $R_a$ and $R_b$, and need to validate that
+ $\frac{R_b}{R_a}==\frac{R_b'}{R_a'}$, with some margin of error (0.01%).
+
+If this is false, revert. Otherwise, calculate the scale of $V_a$ and $V_b$, which is $scale = \frac{R_a}{R_a'}$. Therefore, $V_a = scaleV_a'$ and $V_b = scaleV_b'$.
+
+Finally, ensure the price is close enough to the target price, and the centeredness is above margin.
+
+### On Swap
+
+When executing onSwap, three steps must be performed:
+
+1. Calculate $Q_{0_{current}}$. If `blockTimestamp > endQ0Time`, return $Q_{0_{target}}$. Else, calculate $Q_{0_{current}}$ based on the formula:
+    
+    $\sqrt{Q_{0_{current}}} = \sqrt{Q_{0_{initial}}}(\frac{\sqrt{Q_{0_{target}}}}{\sqrt{Q_{0_{initial}}}})^{\frac{blockTimestamp - startQ0Time}{endQ0Time - startQ0Time}}$
+    
+2. Check whether Q0 (the price range) is updating. If so, adjust the virtual balances as follows, keeping the pool centeredness constant. This won't move the pool `OUT OF RANGE` if there's no swap, which makes off-chain calculations more reliable and optimizes the price interval calculation when the pool is `OUT OF RANGE`.
```

Comment:
This part will probably change, since now we will always update the virtual balances when a swap happens.

Reply:
Ok; Will remove the conditional part, and we can review this again when we have the final code.